File: common/src/main/java/com/imsproject/common/dataAccess/abstracts/CounterDAO.java
package com.imsproject.common.dataAccess.abstracts;


import com.imsproject.common.dataAccess.DaoException;

public interface CounterDAO {

    Integer selectCounter() throws DaoException;

    void insertCounter(Integer value) throws DaoException;

    void incrementCounter() throws DaoException;

    void resetCounter() throws DaoException;
}

File: common/src/main/java/com/imsproject/common/dataAccess/abstracts/CounterDAO.kt
package com.imsproject.common.dataAccess.abstracts

import com.imsproject.common.dataAccess.DaoException


interface CounterDAO {
    @Throws(DaoException::class)
    fun selectCounter(): Int

    @Throws(DaoException::class)
    fun insertCounter(value: Int)

    @Throws(DaoException::class)
    fun incrementCounter()

    @Throws(DaoException::class)
    fun resetCounter()
}
